x-pack/plugins/apm/public/components/alerting/rule_types/anomaly_rule_type/select_anomaly_detector.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiSelectable, EuiSelectableOption } from '@elastic/eui';
+import {
+  AnomalyAlertDetectorType,
+  ANOMALY_ALERT_DETECTORS,
+} from '../../../../../common/rules/apm_rule_types';
+
+interface Props {
+  values: AnomalyAlertDetectorType[];
+  onChange: (values?: AnomalyAlertDetectorType[]) => void;
+}
+
+export function SelectAnomalyDetector({ values, onChange }: Props) {
+  const options: EuiSelectableOption[] = ANOMALY_ALERT_DETECTORS.map(
+    (option) => ({
+      key: option.type,
+      label: option.label,
+      checked: values.includes(option.type) ? 'on' : undefined,
+    })
+  );
+
+  const onOptionSelect = useCallback(
+    (selectedOptions: EuiSelectableOption[]) => {
```

Comment:
Can you try this?
```suggestion
    (selectedOptions: EuiSelectableOption<{key: AnomalyAlertDetectorType}>) => {
```

Reply:
selectedOptions still needs to be an array of objects

Reply:
> selectedOptions still needs to be an array of objects

Sure, that was a typo